lib/style-transform.js

```diff
@@ -1,407 +1,126 @@
-// based on Stylis (MIT)
-// modified by Guillermo Rauch to add support for custom
-// attributes instead of custom selector prefix
-/*!
- *
- *          __        ___
- *    _____/ /___  __/ (_)____
- *   / ___/ __/ / / / / / ___/
- *  (__  ) /_/ /_/ / / (__  )
- * /____/\__/\__, /_/_/____/
- *          /____/
- *
- * stylis is a small css compiler
- *
- * @licence MIT
- */
-(function (factory) {
-    if (typeof exports === 'object' && typeof module !== 'undefined') {
-        module.exports = factory(global);
-    } else if (typeof define === 'function' && define.amd) {
-        define(factory(window));
-    } else {
-        window.stylis = factory(window);
+const stylis = require('stylis')
+
+function middleware(ctx, str, line, col, prefix) {
+  // ctx 1.5 is a selector
+  if (ctx !== 1.5) {
+    return
+  }
+
+  // Stylis adds the namespace at the beginning
+  if (str.indexOf(prefix + ' ') !== 0) {
+    return
+  }
+
+  // while want to append it to selectors.
+  // Strip stylis' namespace.
+  str = str.substring(prefix.length).trim();
+
+  var i = 0
```

Comment:
@giuseppeg Apart from `[title="hello world"] h2` an attribute case what other case do you think that will break on. I have the following that will make it work in the above case.

```js
if (pos !== -1) {
	var edge;

	if (/\[.*[ ].*\]/g.exec(str) !== null) {
		edge = true;
		str = str.replace(/ (.*(?=\]))/g, '<-!->$1');
	}

	var parts = str.split(' ');

	for (var i = 0; i < parts.length; i++) {
		var part = parts[i];
		pos = part.indexOf(':');

		parts[i] = pos !== -1 ? part.substring(0, pos) + prefix + part.substring(pos) : part + prefix;
	}

	str = parts.join(' ');

	if (edge === true) {
		str = str.replace(/<-!->/g, ' ');
	}

	return str;
}
```